[TOW-288][Detail] 운동하는 법 보기 구현

jiyeonO opened the pull request:
- Youtube 연동 (앱이 존재한다면 앱 열기, 없다면 웹 연동)
- [Jira](https://davidyoon1122.atlassian.net/browse/TOW-288?atlOrigin=eyJpIjoiZjE4MjAwMDFkZTVjNGY3YjhjYWIwYWQ3MTM1ZWVhZjQiLCJwIjoiaiJ9)

Commit:
[TOW-288][Detail] 운동하는 법 보기 구현
- Youtube 연동 (앱이 있다면 앱, 없다면 웹 연동)

davidyoon891122 commented on TodayWod/Features/Home/WorkOut/Detail/Row/WodFeature.swift:
```diff
@@ -43,6 +44,9 @@ struct WodFeature {
             switch action {
+            case .didTapOpenYoutube:
+                ApplicationLoader.open(type: .youtube(query: state.model.title))
+                return .none
```
didTapOpenYoutube 액션은 어떻게 테스트 하면 좋을까요 ?

jiyeonO replied:
하는 김에 TC 추가했습니다~ 반영 완료 ac5db8b05ec634d35c1eb8e75cd45aee852cf5d1

davidyoon891122 commented on TodayWod/Common/ApplicationLoader.swift:
```diff
@@ -0,0 +1,55 @@
+}
+
+class ApplicationLoader {
+    
```
아래 질문과 같은 맥락인데, 주입 시키는 방향은 어떨까 해서요 ?

jiyeonO replied:
client로 변경 완료했습니다 ac5db8b05ec634d35c1eb8e75cd45aee852cf5d1

Commit:
Merge branch 'develop' into feature/TOW-288

Commit:
- WodFeature에 FeatureAction 적용 리팩토링

* 리뷰 반영
- ApplicationLoader를 Client로 변경
- WodFeature TC 추가

Commit:
리뷰 반영
- async 방식에서 completion nil로 변경
- 주의 : async 사용으로 인해 .run 함수 안에서 호출하면 메인스레드 진행이 보장되지 않는 문제 발생

davidyoon891122 reviewed (approved):
확인하였습니다.
